src/crypto/shelley/catalystUtils.js

```diff
@@ -0,0 +1,84 @@
+// properties must have quotes for bindings
+/* eslint-disable quote-props */
+/* eslint-disable camelcase */
+// // @flow
+
+import {
+  Bip32PrivateKey,
+  encode_json_str_to_metadatum,
+  MetadataJsonSchema,
+  GeneralTransactionMetadata,
+  BigNum,
+} from '@emurgo/react-native-haskell-shelley'
+import {generateMnemonic, mnemonicToEntropy} from 'bip39'
+
+export const CatalystLabels = Object.freeze({
+  DATA: 61284,
+  SIG: 61285,
+})
+export function generateRegistration(request: {|
```

Comment:
Heads-up: all the functions here should be `async` and should return `Promise<T>`. Basically all calls to `'@emurgo/react-native-haskell-shelley` API are async, so you'll need to add a few awaits in here. Note also that since the library is incomplete, we don't have flow types (we could, but flow would throw many errors for the types that are missing).